app/scripts/views/sign_up.js

```diff
@@ -51,6 +51,17 @@ function (Cocktail, _, p, BaseView, FormView, Template, Session, AuthErrors,
         return p(false);
       }
 
+      /**
+       * check if there is an account we can suggest for an OAuth flow.
+       */
+      var account = this.user.getChooserAccount();
+      var serviceName = this.relier.get('serviceName');
+
+      if (account.get('email') && serviceName) {
```

Comment:
You can use `account.isEmpty()` rather than checking the email and `this.relier.isDirectAccess()` to check if we're in a relier flow or not.